components/monitoring/os/os.js

```diff
@@ -17,70 +17,38 @@
 */
 "use strict";
 
-// Storing static data to call si library less often
-let osData = {};
-let versionData = {};
-let userData = [];
-const programData = "";
+// Storing static data to call libraries less often
+// change to let if we add user refresh functionality
+const osData = si.osInfo();
+const versionData = si.versions();
+const userData = si.users();
+const programData = plistr.getProgs();
 
 /**
 * Called once to initiate the page
 */
 function initOs() {
-    // Loads OS data
-    if ($.isEmptyObject(osData)) {
-        si.osInfo()
-            .then(data => {
-                osData = data;
-                $("#subtitle").text(osData.distro);
-                $("#os-container").append(osHtml(osData));
-            });
-    } else {
-        $("#subtitle").text(osData.distro);
-        $("#os-container").append(osHtml(osData));
-    }
-    // Loads version data
-    if ($.isEmptyObject(versionData)) {
-        si.versions()
-            .then(data => {
-                versionData = data;
-                $("#versions-container").html(versionsHtml(versionData));
-            });
-    } else {
-        $("#versions-container").html(versionsHtml(versionData));
-    }
-    // Loads user data
-    if (userData.length === 0) {
-        si.users()
-            .then(data => {
-                userData = data;
-                $("#user-container").html(userHtml(userData));
-            });
-    } else {
-        $("#user-container").html(userHtml(userData));
-    }
-    getPrograms();
-}
-
-function getPrograms() {
-    let html = "";
-    plistr.getProgs()
-        .then(data => {
-            data.forEach(program => {
-                html += `<tr><td>${program.name}</td><td>${program.version}</td></tr>`;
-            });
-            $("#loading").remove();
-            document.querySelector("#table-head").style.display = "";
-            $("#programs-container").html(html);
-        })
-        .catch(error => {
-            $("#loading").remove();
-            document.querySelector("#table-head").style.display = "";
-            $("#programs-container").html(error);
-        });
+    // Renders OS data once ready
+    osData.then(data => {
+        $("#subtitle").text(data.distro);
+        $("#os-container").append(osHtml(data));
+    });
+    // Renders version data once ready
+    versionData.then(data => {
+        $("#versions-container").html(versionHtml(data));
+    });
+    // Renders user data once ready
+    userData.then(data => {
+        $("#user-container").html(userHtml(data));
+    });
+    // Renders program data once ready
+    programData.then(data => {
+        $("#loading").remove();
+        document.querySelector("#table-head").style.display = "";
+        $("#programs-container").html(programHtml(data));
+    });
```

Comment:
I agree with these changes, expect for the removal of the catch from this promise.
I have recently added that, as a fallback for mac users.
Currently, ProgListr has no mac support, and this would result in a console error.
Within a catch block we could show the error to the user,
so there is no confusing why it "keeps loading" for minutes/years.